### PR TITLE
Revise ConvolutionBackpropData operation class

### PR DESCRIPTION
**Pull request by ggalieroc**
Ticket: 46094

- Enhance dynamic shape inference in `validate_and_infer_types` method.
- Enhance node validation checks.
- Add type_prop unit tests.

### CODE REVIEW

File: ngraph/test/type_prop/convolution_backprop_data.cpp [hunk 1]
```diff
@@ -0,0 +1,1113 @@
+//*****************************************************************************
```
Comment (sdurawa) [inline ngraph/test/type_prop/convolution_backprop_data.cpp:1]: Missing type prop tests with setting `output_padding `attribute for v0 and v1.

Comment (ggalieroc) [reply ngraph/test/type_prop/convolution_backprop_data.cpp:1]: Done

**Comment by jdanieck** [comment]
@sdurawa is on sick leave for 2 weeks so don't wait for him to review this PR.

File: ngraph/core/src/op/convolution.cpp [hunk 2]
```diff
@@ -162,27 +162,29 @@ bool op::v1::ConvolutionBackpropData::is_dynamic() const
...
+
     if (data_pshape.rank().is_static())
     {
         shape = PartialShape{vector<Dimension>(data_pshape.rank().get_length() - 2)};
     }
-    else
+    else if (filter_pshape.rank().is_static())
     {
-        shape = PartialShape{vector<Dimension>(m_strides.size())};
+        shape = PartialShape{vector<Dimension>(filter_pshape.rank().get_length() - 2)};
```
[the remark this answers is not in the document]
Comment (ilyachur) [reply ngraph/core/src/op/convolution.cpp:183]: Ok, I understood that you have allowed to have empty strides.

**Review by ilyachur — APPROVED** [review]

File: ngraph/core/src/op/convolution.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (ggalieroc) [reply ngraph/core/src/op/convolution.cpp:183]: Exactly. The call for `get_output_shape()` happens before we check on strides and dilations. If rank of data is not static, then we would rely on size of strides (equal to num of spatial dimensions) to set the rank of output partial shape. However, if we construct the op with empty strides, then the output rank would be set wrongly.

**Comment by andrei-kochin** [comment]
@nshchego please take a look at this reference implementation from CPU perspective

**Review by sdurawa — APPROVED** [review]

**Comment by ilyachur** [comment]
Please rebase this PR to the latest master branch.

**Pull request merged**